### PR TITLE
 Support webhook and embed on achievements/advancements

### PR DESCRIPTION
**Pull request by JuanDelPueblo**
I worked on this feature for my own server, but I figure others would want this too. Updated the messages.yml for all 10 languages as well. Tested on a local server, appears to work fine.

Example of how it shows up (can update the color to your preference, I borrowed it from the first join message)
image

### CODE REVIEW

File: src/main/resources/messages/en.yml [hunk 1]
```diff
@@ -269,20 +269,36 @@ MinecraftPlayerDeathMessage:
       Text: ""
       IconUrl: ""
     Timestamp: false
-
-# Minecraft -> Discord achievement/advancement messages
 #
 # Available placeholders:
```
Comment (Vankka) [inline src/main/resources/messages/en.yml:273]: This should say "Available placeholders for PlayerAchievement messages:" (and match in other languages, Google translate can be used to translate)

File: src/main/resources/messages/en.yml [hunk 2]
```diff
@@ -269,20 +269,36 @@ MinecraftPlayerDeathMessage:
...
+    Title:
+      Text: ""
+      Url: ""
+    Description: ""
+    Fields: []
+    ImageUrl: ""
+    Footer:
+      Text: ""
+      IconUrl: ""
+    Timestamp: false
```
Comment (Vankka) [inline src/main/resources/messages/en.yml:301]: There should be a empty line after this line (in all languages)

File: src/main/java/github/scarsz/discordsrv/api/events/AchievementMessagePostProcessEvent.java [hunk 3]
```diff
@@ -29,14 +32,29 @@
 
     @Getter private String achievementName;
     @Getter @Setter private String channel;
-    @Getter @Setter private String processedMessage;
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/api/events/AchievementMessagePostProcessEvent.java:32]: the getter and setter for this should be deprecated, see https://github.com/DiscordSRV/DiscordSRV/blob/develop/src/main/java/github/scarsz/discordsrv/api/events/DeathMessagePostProcessEvent.java#L60-L68

File: src/main/java/github/scarsz/discordsrv/api/events/AchievementMessagePreProcessEvent.java [hunk 4]
```diff
@@ -29,12 +30,22 @@
 
     @Getter @Setter private String achievementName;
     @Getter @Setter private String channel;
-    @Getter @Setter private String message;
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/api/events/AchievementMessagePreProcessEvent.java:32]: the getter and setter for this should be deprecated, see https://github.com/DiscordSRV/DiscordSRV/blob/develop/src/main/java/github/scarsz/discordsrv/api/events/DeathMessagePreProcessEvent.java#L52-L62

File: src/main/java/github/scarsz/discordsrv/listeners/PlayerAchievementsListener.java [hunk 5]
```diff
@@ -55,40 +56,77 @@ public void onPlayerAchievementAwarded(PlayerAchievementAwardedEvent event) {
...
+        Message discordMessage = DiscordSRV.getPlugin().translateMessage(messageFormat, translator);
+        if (discordMessage == null) return;
+
+        webhookName = translator.apply(webhookName, true);
+        webhookAvatarUrl = translator.apply(webhookAvatarUrl, true);
+
+        if (achievementName.length() < 3) {
+            DiscordSRV.debug("Not sending achievement message, because the achievement name is less than three characters long. Message: " + messageFormat);
+            return;
+        }
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/listeners/PlayerAchievementsListener.java:111]: This check isn't necessary for achievements/advancements

File: src/main/java/github/scarsz/discordsrv/listeners/PlayerAdvancementDoneListener.java [hunk 6]
```diff
@@ -65,46 +66,82 @@ public void onPlayerAdvancementDone(PlayerAdvancementDoneEvent event) {
...
+        Message discordMessage = DiscordSRV.getPlugin().translateMessage(messageFormat, translator);
+        if (discordMessage == null) return;
+
+        webhookName = translator.apply(webhookName, true);
+        webhookAvatarUrl = translator.apply(webhookAvatarUrl, true);
+
+        if (advancementName.length() < 3) {
+            DiscordSRV.debug("Not sending achievement message, because it's less than three characters long. Message: " + messageFormat);
+            return;
+        }
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/listeners/PlayerAdvancementDoneListener.java:127]: This check isn't necessary for achievements/advancements

File: src/main/resources/messages/en.yml (hunk 1 above)
Comment (JuanDelPueblo) [reply src/main/resources/messages/en.yml]: Each messages.yml should now have the same line translated

File: src/main/resources/messages/en.yml (hunk 2 above)
Comment (JuanDelPueblo) [reply src/main/resources/messages/en.yml:301]: Added to all messages.yml languages

File: src/main/java/github/scarsz/discordsrv/listeners/PlayerAchievementsListener.java (hunk 5 above)
Comment (JuanDelPueblo) [reply src/main/java/github/scarsz/discordsrv/listeners/PlayerAchievementsListener.java]: Removed

File: src/main/java/github/scarsz/discordsrv/listeners/PlayerAdvancementDoneListener.java (hunk 6 above)
Comment (JuanDelPueblo) [reply src/main/java/github/scarsz/discordsrv/listeners/PlayerAdvancementDoneListener.java]: Removed

File: src/main/java/github/scarsz/discordsrv/api/events/AchievementMessagePostProcessEvent.java (hunk 3 above)
Comment (JuanDelPueblo) [reply src/main/java/github/scarsz/discordsrv/api/events/AchievementMessagePostProcessEvent.java:32]: Should be added, let me know if it is done correctly

File: src/main/java/github/scarsz/discordsrv/api/events/AchievementMessagePreProcessEvent.java (hunk 4 above)
Comment (JuanDelPueblo) [reply src/main/java/github/scarsz/discordsrv/api/events/AchievementMessagePreProcessEvent.java:32]: Should be added, let me know if it is done correctly

**Pull request merged**